common4j/src/main/com/microsoft/identity/common/java/controllers/CommandDispatcher.java

```diff
@@ -491,9 +487,25 @@ private static CommandResult executeCommand(@SuppressWarnings(WarningType.rawtyp
             }
         }
 
+        // set correlation id on Local Authentication Result
+        setCorrelationIdOnResult(commandResult, correlationId);
+        setTelemetryOnResult(commandResult, correlationId);
         return commandResult;
     }
 
+    private static void setTelemetryOnResult(@NonNull final CommandResult commandResult,
+                                             @NonNull final String correlationId) {
+        final List<Map<String, String>> telemetryMap = Telemetry.getInstance().getMap(
+                correlationId
+        );
+
+        if (commandResult.getResult() instanceof LocalAuthenticationResult) {
+            ((LocalAuthenticationResult) commandResult.getResult()).setTelemetry(telemetryMap);
+        } else if (commandResult.getResult() instanceof BaseException) {
+            ((BaseException) commandResult.getResult()).setTelemetry(telemetryMap);
+        }
```

Comment:
adding log for some other type of results?

Reply:
yeah, you should do something in the other else case.

Reply:
Something like... unless this actually happens all the time.
```suggestion
        } else if (commandResult.getResult() != null) {
          Logger.warn(TAG + ":setTelemetryOnResult", "Unknown result type " + commandResult.getResult().getCanonicalName());
        }
```

Reply:
It would happen frequently in MSAL..because we have LoadAccountCommand, RemoveAccountCommand, DeviceModeCommand and their corresponding results coming through. So I don't think we need warn log here...we could do a `verbose` level log if we still want to know that. 

Ideally we also put telemetry on results for these other commands, however it is irrelevant right now for broker because broker getAccounts, removeAccount operations are actually not using commands and the dispatcher. It's only used by MSAL, and since my telemetry work was specific to broker, therefore I didn't bother about that, but at some point we need to be consistent with MSAL and Broker and probably start using Dispatcher in the same way in each operation.

Reply:
Added verbose level log. Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1636/commits/4f18eff7cf4b93131c2ebd4ec44908688f0efe0b